.github/workflows/publish_release.yml

```diff
@@ -0,0 +1,89 @@
+name: Publish the latest release on PyPI
+
+on:
+  push:
+    tags:
+      - 'v*'
+    branches:
+      - main  # Restricts the workflow to only run on tags pushed to the main branch
+
+jobs:
+  validate-tag:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Validate Tag Format
+        run: |
+          TAG_REGEX='^v[0-9]+\.[0-9]+\.[0-9]+(-[a-zA-Z0-9\-\.]+)?$'
+          echo "Regex to match: $TAG_REGEX"
+          TAG="${GITHUB_REF/refs\/tags\//}"
+          echo "Tag extracted: $TAG"
+          if [[ "$TAG" =~ $TAG_REGEX ]]; then
+            echo "Tag format is valid."
+          else
+            echo "::error::Tag $TAG does not match the 'vMAJOR.MINOR.PATCH' or 'vMAJOR.MINOR.PATCH-pre-release' format."
+            exit 1
+          fi
+        shell: bash
+
+  build-and-package:
+    needs: validate-tag
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.11'
```

Comment:
also here we should use `poetry install --no-dev`

And we can make sure all tests are run successfully before the build and publish as another safety

Reply:
There is a kind of dry publish. I used it in a test file, it works. So we can also add that simulation test before publish